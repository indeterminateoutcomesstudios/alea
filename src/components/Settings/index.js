// @flow
import React from 'react'
import { connect } from 'react-redux'
import Modal from 'components/Modal'
import Logout from 'components/Settings/Logout'
import Name from 'components/Settings/Name'
import ThemeSwitcher from 'components/Settings/ThemeSwitcher'
import type { Theme } from 'styles/themes'
import type { StateProps, DispatchProps } from 'containers/Settings'

type Props = {
  theme: Theme
} & StateProps &
  DispatchProps

const Settings = (props: Props) => {
  const {
    displayName,
    changeDisplayName,
    theme,
    changeTheme,
    logout,
    dismissSettings
  } = props

  return (
    <Modal dismiss={dismissSettings}>
      {() => (
        <div>
          <Name name={displayName} onChange={changeDisplayName} />

          <ThemeSwitcher currentTheme={theme.name} changeTheme={changeTheme} />

          <Logout logout={logout} />
        </div>
      )}
    </Modal>
  )
}

export default Settings
