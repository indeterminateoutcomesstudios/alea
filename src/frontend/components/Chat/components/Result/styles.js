// @flow
import styled from 'styled-components'
import { colors } from 'frontend/styles/common'

export const Container = styled.div`
  margin-top: 0.6rem;
  padding: 0;
  background: ${props => props.theme.backgroundSecondary};
  border-left: 3px solid ${colors.blue};
  padding-left: 5px;
  border-radius: 5px;
`

export const Row = styled.div`
  border-top: 1px solid ${props => props.theme.borderColor};

  &:first-child {
    border-top: 0;
  }
`

export const Modifier = styled.span`
  font-weight: 400;
  color: ${props => props.theme.colorSecondary};
  margin-right: 0.4em;
`

export const Roll = styled.span`
  font-weight: 600;
  margin-right: 0.4em;
`
