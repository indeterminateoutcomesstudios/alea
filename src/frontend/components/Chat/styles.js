// @flow
import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${props => props.theme.background};
  width: 320px;
  overflow: hidden;
  display: flex;
  flex-direction: column;

  ${props =>
    props.isPinned
      ? `
    position: absolute;
    top: 4rem;
    right: 0;
    bottom: 0;
  `
      : `
    position: absolute;
    top: auto;
    right: 1rem;
    bottom: 1rem;
    box-shadow: 0 3px 5px 0 rgba(0, 0, 0, 0.2);
    border-radius: 5px;
  `};
`

export const Loading = styled.div`
  padding: 10px;
`

export const Error = Loading

export const MessagesWrapper = styled.div`
  min-height: 0;
  flex: 1;
`
export const Messages = styled.div`
  overflow-y: ${props => (props.isPinned ? 'scroll' : 'hidden')};
  overflow-x: hidden;
  width: 315px;
  height: 100%;

  display: flex;
  flex-direction: column;
  flex: ${props => (props.isPinned ? '1' : 'none')};
`
