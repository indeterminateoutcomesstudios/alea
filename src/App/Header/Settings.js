/* @flow */
import React from 'react'
import { connect } from 'react-redux'
import * as colors from 'styles/colors'
import styled from 'styled-components'

type Props = {
  showSettings: Function
}

const Settings = (props: Props) => {
  const { showSettings } = props

  const Button = styled.div`
    width: 4.8rem;
    height: 4.8rem;
    text-align: center;
    line-height: 4.8rem;

    &:hover {
      cursor: pointer;
      background-color: ${colors.lightBlue};
    }
  `

  return (
    <Button onClick={showSettings}>
      <i className={`fa fa-cog`}></i>
    </Button>
  )
}

const mapDispatchToProps = (dispatch: Function) => ({
  showSettings: () => dispatch({ type: 'SHOW_SETTINGS' })
})

export default connect(null, mapDispatchToProps)(Settings)
