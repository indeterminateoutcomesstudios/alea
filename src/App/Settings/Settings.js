/* @flow */
import React from 'react'
import { connect } from 'react-redux'
import { compose, withHandlers } from 'recompose'
import styled from 'styled-components'
import withTheme from 'containers/withTheme'
import type { ThemeProps } from 'containers/withTheme'
import Name from './Name'
import ThemeSwitcher from './ThemeSwitcher'
import Logout from './Logout'
import { changeTheme, changeDisplayName } from 'actions'
import * as themes from 'styles/themes'
import type { State } from 'store'

const withInteractions = compose(
  withTheme,
  connect(
    (state: State) => ({
      displayName: state.user.profile.displayName,
      showSettings: state.ui.showSettings
    }),
    (dispatch: Function) => ({
      changeTheme: (theme: Theme) => dispatch(changeTheme(theme)),
      changeDisplayName: (name: string) => dispatch(changeDisplayName(name)),
      logout: () => dispatch({ type: 'LOGOUT' }),
      hideSettings: () => dispatch({ type: 'HIDE_SETTINGS' })
    })
  ),
  withHandlers({
    changeTheme: props => theme => props.changeTheme(theme),
    updateProfileName: props => name => props.changeDisplayName(name),
    logout: props => props.logout,
    handleWrapperClick: props => props.hideSettings,
    handleInnerClick: () => e => e.stopPropagation()
  })
)

type Props = {
  displayName: string,
  changeTheme: Function,
  changeDisplayName: Function,
  logout: Function,
  hideSettings: Function
} & ThemeProps

const SettingsContainer = withTheme(styled.div`
  display: flex;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: rgba(0, 0, 0,
    ${props => props.theme === 'light' ? '0.5' : '0.7'}
  );
`)

const SettingsInner = withTheme(styled.div`
  display: flex;
  flex-direction: column;
  width: 300px;
  margin: auto;
  border-radius: 5px;
  padding: 2.4rem;
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.3);
  background-color: ${props => props.theme == 'light'
    ? 'white'
    : themes[props.theme].backgroundSecondary
  };
`)

const Settings = (props: Props) => (
  <SettingsContainer onClick={props.handleWrapperClick}>
    <SettingsInner onClick={props.handleInnerClick}>
      <Name name={props.displayName} onChange={props.updateProfileName} />

      <ThemeSwitcher
        currentTheme={props.theme}
        changeTheme={props.changeTheme}
      />

      <Logout logout={props.logout}/>
    </SettingsInner>
  </SettingsContainer>
)

export default withInteractions(Settings)
