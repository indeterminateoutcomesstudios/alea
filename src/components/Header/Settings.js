/* @flow */
import React from 'react'
import { colors } from 'styles/common'
import styled from 'styled-components'

const Button = styled.div`
  width: 4.8rem;
  height: 4.8rem;
  text-align: center;
  line-height: 4.8rem;

  &:hover {
    cursor: pointer;
    background-color: ${colors.lightBlue};
  }
`

type Props = {
  showSettings: Function
}

const Settings = (props: Props) => {
  const { showSettings } = props

  return (
    <Button onClick={showSettings}>
      <i className={`fa fa-cog`} />
    </Button>
  )
}

export default Settings
