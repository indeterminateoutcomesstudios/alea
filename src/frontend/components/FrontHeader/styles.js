// @flow
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { fonts } from 'frontend/styles/common'

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  color: white;
  height: 8rem;
`

export const LogoButton = styled(Link)`
  padding-top: 27px;
  padding-left: 2rem;
  text-decoration: none;
  color: white;
`

export const Button = styled(Link)`
  background-color: white;
  border-radius: 5px;
  height: 4rem;
  line-height: 4rem;
  padding: 0 2rem;
  margin: 2rem;
  margin-left: auto;
  color: ${props => props.theme.backgroundInverted};
  font-family: ${fonts.heading};
  font-weight: 700;
  text-decoration: none;
`
