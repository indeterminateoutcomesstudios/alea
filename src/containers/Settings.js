/* @flow */
import React from 'react'
import { connect } from 'react-redux'
import { withTheme } from 'styled-components'
import Settings from 'components/Settings'
import { changeTheme, changeDisplayName } from 'actions'
import type { ThemeName } from 'types'
import type { State } from 'store'

export type StateProps = {
  displayName: string,
  showSettings: boolean
}
const mapStateToProps = (state: State): StateProps => ({
  displayName: state.user.profile.displayName,
  showSettings: state.ui.showSettings
})

export type DispatchProps = {
  changeTheme: (theme: ThemeName) => void,
  changeDisplayName: (name: string) => void,
  logout: Function,
  dismissSettings: Function
}
const mapDispatchToProps = (dispatch: Function): DispatchProps => ({
  changeTheme: (theme: ThemeName) => dispatch(changeTheme(theme)),
  changeDisplayName: (name: string) => dispatch(changeDisplayName(name)),
  logout: () => dispatch({ type: 'LOGOUT' }),
  dismissSettings: () => dispatch({ type: 'HIDE_SETTINGS' })
})

export default connect(mapStateToProps, mapDispatchToProps)(withTheme(Settings))
