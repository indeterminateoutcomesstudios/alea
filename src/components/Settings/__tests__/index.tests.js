// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import Settings from '../index.js'
import { light } from '../../../styles/themes'

describe('Settings component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={light}>
          <Settings
            showSettings={true}
            displayName="test"
            changeDisplayName={() => {}}
            theme={light}
            changeTheme={() => {}}
            logout={() => {}}
            dismissSettings={() => {}}
          />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
