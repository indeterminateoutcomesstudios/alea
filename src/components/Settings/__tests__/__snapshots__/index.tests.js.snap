// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders correctly 1`] = `
<div
  className="sc-bdVaJa brBhVF"
>
  <div
    className="sc-bwzfXH eXaRNR"
  >
    <div>
      <div>
        <label
          className="sc-ifAKCX iyagNL"
        >
          Display Name:
        </label>
        <div
          className="sc-gzVnrw hFWLX"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          test
        </div>
      </div>
      <div>
        <label
          className="sc-ifAKCX iyagNL"
        >
          Theme
        </label>
        <div
          className="sc-iwsKbI SIqHi"
        >
          <button
            className="sc-htpNat cwhfPC"
            onClick={[Function]}
          >
            Light
          </button>
          <button
            className="sc-dnqmqq gPRboo"
            onClick={[Function]}
          >
            Dark
          </button>
        </div>
      </div>
      <div
        className="sc-bxivhb bIqXCY"
      >
        <button
          className="sc-htpNat hnNRQd"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;
